UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/GeneratorImpl.cs

```diff
@@ -104,74 +103,35 @@ public VectorObservationGenerator(ITensorAllocator allocator)
         }
     }
 
-    /// <summary>
-    /// Generates the Tensor corresponding to the Recurrent input : Will be a two
-    /// dimensional float array of dimension [batchSize x memorySize].
-    /// It will use the Memory data contained in the agentInfo to fill the data
-    /// of the tensor.
-    /// </summary>
-    public class RecurrentInputGenerator : TensorGenerator.IGenerator
-    {
-        private readonly ITensorAllocator m_Allocator;
-
-        public RecurrentInputGenerator(ITensorAllocator allocator)
-        {
-            m_Allocator = allocator;
-        }
-
-        public void Generate(
-            TensorProxy tensorProxy, int batchSize, IEnumerable<Agent> agents)
-        {
-            TensorUtils.ResizeTensor(tensorProxy, batchSize, m_Allocator);
-
-            var memorySize = tensorProxy.shape[tensorProxy.shape.Length - 1];
-            var agentIndex = 0;
-            foreach (var agent in agents)
-            {
-                var info = agent.Info;
-                var memory = info.memories;
-                if (memory == null)
-                {
-                    agentIndex++;
-                    continue;
-                }
-                for (var j = 0; j < Math.Min(memorySize, memory.Count); j++)
-                {
-                    if (j >= memory.Count)
-                    {
-                        break;
-                    }
-                    tensorProxy.data[agentIndex, j] = memory[j];
-                }
-                agentIndex++;
-            }
-        }
-    }
 
     public class BarracudaRecurrentInputGenerator : TensorGenerator.IGenerator
     {
         private int m_MemoriesCount;
         private readonly int m_MemoryIndex;
         private readonly ITensorAllocator m_Allocator;
 
-        public BarracudaRecurrentInputGenerator(int memoryIndex, ITensorAllocator allocator)
+        private Dictionary<int, List<float>> m_Memories;
+
+        public BarracudaRecurrentInputGenerator(
+            int memoryIndex,
+            ITensorAllocator allocator,
+            ref Dictionary<int, List<float>> memories)
```

Comment:
why `ref` here?  You only need that if you wanted to change the pointer to `memories` parameter in this constructor.

Reply:
I was making experiments and forgot to remove it.